src/components/inputs/__tests__/InputField.test.tsx

```diff
@@ -0,0 +1,100 @@
+import { render, screen, fireEvent } from "@testing-library/react";
+import { useForm, FormProvider } from "react-hook-form";
+import { describe, it, expect } from "vitest";
+import InputField, { INPUT_FIELD_VARIANTS } from "../InputField";
+import { Button } from "@mui/material";
+import { ReactNode } from "react";
+import userEvent from "@testing-library/user-event";
+import { ERRORS } from "../../../utils/strings";
+
+const renderWithForm = (ui: JSX.Element) => {
+  const Wrapper = ({ children }: { children: ReactNode }) => {
+    const methods = useForm();
+
+    //function created just to trigger useForm's validation
+    const mockSubmitFunction = (data: any) => {
+      console.log(data); //
+    };
+    const { handleSubmit } = methods;
+
+    return (
+      <FormProvider {...methods}>
+        <form onSubmit={handleSubmit(mockSubmitFunction)}>
+          {children}
+          <Button type="submit">Submit</Button>
+        </form>
+      </FormProvider>
+    );
+  };
+
+  return render(ui, { wrapper: Wrapper });
+};
+
+describe("Password InputField component tests", () => {
+  let submitBtn: HTMLButtonElement;
+  let passwordInput: HTMLInputElement;
+
+  beforeEach(() => {
+    renderWithForm(
+      <InputField
+        dataName="password"
+        label="Password"
+        type="password"
+        variant={INPUT_FIELD_VARIANTS.OUTLINED}
+        required
+        minPasswordLength={10}
+      />
+    );
+
+    submitBtn = screen.getByText("Submit");
```

Comment:
Please better use the role for button

```suggestion
    submitBtn = screen.getByRole("button", { name: "Submit" });
```